app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1168,6 +1180,98 @@ class MessagingManager @Inject constructor(
             return@withContext image
         }
 
+    private suspend fun handleVideo(
+        builder: NotificationCompat.Builder,
+        data: Map<String, String>
+    ) {
+        data[VIDEO_URL]?.let {
+            val url = UrlHandler.handle(urlUseCase.getUrl(), it)
+            getVideoFrames(url, !UrlHandler.isAbsoluteUrl(it))?.let { frames ->
+                RemoteViews(context.packageName, R.layout.view_image_flipper).let { remoteViewFlipper ->
+                    if (frames.isNotEmpty()) {
+                        frames.forEach { frame ->
+                            remoteViewFlipper.addView(
+                                R.id.frame_flipper,
+                                RemoteViews(context.packageName, R.layout.view_single_frame).apply {
+                                    setImageViewBitmap(
+                                        R.id.frame,
+                                        frame
+                                    )
+                                }
+                            )
+                        }
+
+                        data[TITLE]?.let { rawTitle ->
+                            remoteViewFlipper.setTextViewText(R.id.title, rawTitle)
+                        }
+
+                        data[MESSAGE]?.let { rawMessage ->
+                            remoteViewFlipper.setTextViewText(R.id.info, rawMessage)
+                        }
```

Comment:
Doesn't the `handleText` function handle this?

Reply:
No, when you set the style as `DecoreatedCustomViewStyle` you actually control the entire notification viewport so the title and text part from the builder is overwritten / not included and the custom view needs to include it itself.

Reply:
To clarify on that last point, the docs here: https://developer.android.com/reference/android/app/Notification.DecoratedCustomViewStyle show how you can override specific parts of the notification. In this case I am overriding the bigContentView so the heads up view and the standard content view will use the `handleText` part but when the notification is expanded it needs to be handled by the custom view here.